pkg/controller/endpointslice/reconciler.go

```diff
@@ -251,6 +258,28 @@ func (r *reconciler) reconcileByAddressType(service *corev1.Service, pods []*cor
 	return r.finalize(service, slicesToCreate, slicesToUpdate, slicesToDelete, triggerTime)
 }
 
+// sliceEqual returns if two slices are equal.  It ignores everything in ObjectMeta besides Labels.
```

Comment:
we are setting more fields in ObjectMeta, why those are not taking into account for equality?

https://github.com/kubernetes/kubernetes/blob/ea978cbe996a7eaf9a50a01534c5196f49d4906a/pkg/controller/endpointslice/utils.go#L137-L154

Reply:
Good point, in this case maybe we can safely compare everything except for Name/GenerateName

Reply:
Pushed an update to the comparison logic and renamed the func since it is specific to the placeholder slice